core/bubbles/mini_workspace_bubble.ts

```diff
@@ -0,0 +1,199 @@
+  ) {
+    super(workspace, anchor, ownerRect);
+    const options = new Options(workspaceOptions);
+    this.validateWorkspaceOptions(options);
+
+    this.svgDialog = dom.createSvgElement(
+      Svg.SVG,
+      {
+        'x': Bubble.BORDER_WIDTH,
+        'y': Bubble.BORDER_WIDTH,
+      },
+      this.contentContainer
+    );
+    workspaceOptions.parentWorkspace = this.workspace;
+    this.miniWorkspace = this.newWorkspaceSvg(new Options(workspaceOptions));
+    const background = this.miniWorkspace.createDom('blocklyMutatorBackground');
+    this.svgDialog.appendChild(background);
+    if (options.languageTree) {
+      background.insertBefore(
+        this.miniWorkspace.addFlyout(Svg.G),
+        this.miniWorkspace.getCanvas()
+      );
+      const flyout = this.miniWorkspace.getFlyout();
+      flyout?.init(this.miniWorkspace);
+      flyout?.show(options.languageTree);
+    }
+
+    this.miniWorkspace.addChangeListener(this.updateBubbleSize.bind(this));
+    this.miniWorkspace
+      .getFlyout()
+      ?.getWorkspace()
+      ?.addChangeListener(this.updateBubbleSize.bind(this));
+    this.updateBubbleSize();
+  }
+
+  dispose() {
+    this.miniWorkspace.dispose();
+    super.dispose();
+  }
+
+  /** @internal */
+  getWorkspace(): WorkspaceSvg {
+    return this.miniWorkspace;
+  }
+
+  /** Adds a change listener to the mini workspace. */
+  addWorkspaceChangeListener(listener: (e: AbstractEvent) => void) {
+    this.miniWorkspace.addChangeListener(listener);
+  }
+
+  /**
+   * Validates the workspace options to make sure folks aren't trying to
+   * enable things the miniworkspace doesn't support.
+   */
+  private validateWorkspaceOptions(options: Options) {
+    if (options.hasCategories) {
+      throw new Error(
+        'The miniworkspace bubble does not support toolboxes with categories'
+      );
+    }
+    if (options.hasTrashcan) {
+      throw new Error('The miniworkspace bubble does not support trashcans');
+    }
+    if (
+      options.zoomOptions.controls ||
+      options.zoomOptions.wheel ||
+      options.zoomOptions.pinch
+    ) {
+      throw new Error('The miniworkspace bubble does not support zooming');
+    }
+    if (
+      options.moveOptions.scrollbars ||
+      options.moveOptions.wheel ||
+      options.moveOptions.drag
+    ) {
+      throw new Error(
+        'The miniworkspace bubble does not scrolling/moving the workspace'
+      );
+    }
+    if (options.horizontalLayout) {
+      throw new Error(
+        'The miniworkspace bubble does not support horizontal layouts'
+      );
+    }
+  }
+
+  /**
+   * Updates the size of this bubble to account for the size of the
+   * mini workspace.
+   */
+  private updateBubbleSize() {
+    if (this.miniWorkspace.isDragging()) return;
+
+    const currSize = this.getSize();
+    const newSize = this.calculateWorkspaceSize();
+    if (
+      Math.abs(currSize.width - newSize.width) < 10 &&
+      Math.abs(currSize.height - newSize.height) < 10
```

Comment:
Should this delta value (`10`) be configurable, or at least a named constant?  Also, what units is it measured in?  10px seems like a pretty noticeable change.

Reply:
Can add a constant. I don't think the change is necessarily "noticable" because we always have margin around the contents of the workspace. So if we don't resize for <10px there will just be slightly less margin than if we do resize.

Reply:
Isn't the border only 6px?  Or am I confused?

Reply:
The border is 6px, but we also have white space that exists around the workspace, which is what I meant by margin (sorry we don't really have language for this stuff so I'm just making it up on the spot lol). 
![image](https://github.com/google/blockly/assets/25440652/a1e6f10a-19d6-46c5-aefb-c4ab097d2286)